docs/styleguide/PyGuide.md

```diff
@@ -816,3 +816,18 @@ This is required so that the autogenerated API documentation is rendered properl
 
 If the autogenerated API documentation does not show type hints for certain arguments despite the fact that the type hints are present in the object's implementation code,
 or if the type hints do not refer to the API symbol's canonical alias, then the type hint should be explicitly declared in the docstring using the `:type *param_name*:` directive (or `:rtype:` for return types).
+
+<a id="s5-test-suite-coding-rules"></a>
+<a id="5-test-suite-coding-rules"></a>
+<a id="test-suite-coding-rules"></a>
+
+## 5 Test suite coding rules
+
+Unit tests are written and executed using the [pytest](https://pytest.org) framework.
+
+*to_be_extended_with_more_rules_capturing_the_current_style*
```

Comment:
Some suggestion to this:
- use pathlib.Path for fixtures for paths. Now it's all return str
- Add fixtures to scripts wehere it's used, and avoid add to conftest.py only it's common fixtures that can used in many tests.
- tests does not contain logic to manual comparing results as current e2e. 
- Used docstring and type hints for any fixtures
- Add best practics for rare case like colleting data from the tests ( like https://github.com/openvinotoolkit/nncf/pull/2152#discussion_r1373915533) 
- Add bad practics like this https://github.com/openvinotoolkit/nncf/blob/76bb1f085ef21fe000ba3726115bf4c84d6ac937/tests/tensorflow/conftest.py#L73 


May be you can analyze our tests to provide more examples of good and bad practice.

Reply:
And will be good add point about add point about temples that used that used for PTQ.
My suggestion to add all templates to `tests/shared/test_templates`.

Reply:
See the updated version. The global variable usage with `pytest.metrics_dump_path` is covered by the guide in section 3.3. Added the pathlib usage to section 3.7. `tests/shared` is actually meant for non-test-case code reused in other test folders, I meant for the actual test code that would be reused across multiple frameworks to be located in `tests/cross_fw`

Reply:
Looks good 👍 
About templates is ok, i will replace templates.